### PR TITLE
fix reverse_graph

### PR DESCRIPTION
**Pull request by matthiasdiener**

### CODE REVIEW

File: pytato/transform.py [hunk 1]
```diff
@@ -1131,6 +1131,10 @@ def reverse_graph(graph: Dict[ArrayOrNames, Set[ArrayOrNames]]) \
     result: Dict[ArrayOrNames, Set[ArrayOrNames]] = {}
 
     for node_key, edges in graph.items():
+        # Make sure every node is in the result even if it has no users
+        if node_key not in result:
+            result[node_key] = set()
```
Comment (inducer) [inline pytato/transform.py:1136]:
```suggestion
        result.setdefault(node_key, set())
```

File: pytato/transform.py [hunk 2]
```diff
@@ -1131,6 +1131,10 @@ def reverse_graph(graph: Dict[ArrayOrNames, Set[ArrayOrNames]]) \
     result: Dict[ArrayOrNames, Set[ArrayOrNames]] = {}
 
     for node_key, edges in graph.items():
+        # Make sure every node is in the result even if it has no users
```
Comment (inducer) [inline pytato/transform.py:1134]: Add a regression test for this.

**Review by inducer — COMMENTED** [review]
LGTM with these two changes.

**Review by kaushikcfd — APPROVED** [review]
Looks good, thanks!

File: pytato/transform.py (hunk 2 above)
Comment (matthiasdiener) [reply pytato/transform.py:1134]: bd8c336

**Pull request merged**